atc/db/resource.go

```diff
@@ -37,28 +37,31 @@ type Resource interface {
 	PinComment() string
 	ResourceConfigCheckError() error
 	ResourceConfigID() int
+	ResourceConfigScopeID() int
 
 	CurrentPinnedVersion() atc.Version
 
 	ResourceConfigVersionID(atc.Version) (int, bool, error)
 	Versions(page Page) ([]atc.ResourceVersion, Pagination, bool, error)
+	SaveUncheckedVersion(atc.Version, ResourceConfigMetadataFields, ResourceConfig, creds.VersionedResourceTypes) (bool, error)
 
 	EnableVersion(rcvID int) error
 	DisableVersion(rcvID int) error
 
 	PinVersion(rcvID int) error
 	UnpinVersion() error
 
-	SetResourceConfig(lager.Logger, atc.Source, creds.VersionedResourceTypes) (ResourceConfig, error)
+	SetResourceConfig(lager.Logger, atc.Source, creds.VersionedResourceTypes) (ResourceConfigScope, error)
 	SetCheckError(error) error
```

Comment:
This should probably move to scopes, because of the following scenario:

* assume two teams have a resource config but with unique version history, i.e. different scopes
* there are no global workers
* team A has workers, team B doesn't
* team B tries to check, but fails because there are no workers, and sets the check error
* team A tries to check, works, clears out error

These two teams would be flipping the error on and off, and team A would see a bogus error.